var/spack/repos/builtin/packages/quantlib/package.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.package import *
+
+
+class Quantlib(CMakePackage):
+    """The QuantLib project is aimed at providing a comprehensive software
+    framework for quantitative finance. QuantLib is a free/open-source library
+    for modeling, trading, and risk management in real-life."""
+
+    homepage = "https://www.quantlib.org/"
+    url = "https://github.com/lballabio/QuantLib/releases/download/v1.31.1/QuantLib-1.31.1.tar.gz"
+
+    maintainers("TheQueasle")
+
+    version("1.31.1", sha256="13b5346217153ae3c185e0c640cc523a1a6522c3a721698b2c255fd9a1a15a68")
+
+    variant("benchmark", default=True, description="Build benchmark")
+    variant("examples", default=True, description="Build examples")
+    variant("test", default=True, description="Build test suite")
```

Comment:
For testing, it is preferred to use `self.run_tests` as the flag in `cmake_args`. More at https://spack.readthedocs.io/en/latest/packaging_guide.html#checking-build-time-test-results, or an example of its use at https://github.com/spack/spack/blob/1ac2ee8043ab802ff5c9161ac35c269f22df3c34/var/spack/repos/builtin/packages/exago/package.py#L219